### PR TITLE
Add Contributing docs, add donate section to README

### LINKED ISSUE #723: Add a contributing guidelines document?
Hi there! 👋 

Thanks for creating a great package - we use it for our [FEC API](https://github.com/fecgov/openFEC). In preparation for PyCon sprints, I was checking out your project and found that there isn't a contributing guide.

Here's the [GitHub info](https://help.github.com/en/articles/setting-guidelines-for-repository-contributors) on why contributing guidelines are helpful - I'd be happy to work on this issue if you'd like! 

Thanks again,
Laura

### PR DESCRIPTION
**Pull request by lbeaufort**
Resolves #723:
- Adds **Contributing** doc, links to it in the README
- Updates README to have a donate section to match Flask and other Pallets projects

Notes:
- The contributing document is based on [Flask's contributing document](https://github.com/pallets/flask/blob/master/CONTRIBUTING.rst)
- The contributing document is slightly different for testing and documentation instructions

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Support questions
+-----------------
+
+Please, don't use the issue tracker for this. Use one of the following
+resources for questions about your own code:
+
+* The IRC channel ``#pocoo`` on FreeNode.
+* The IRC channel ``#python`` on FreeNode for more general questions.
+* The mailing list flask@python.org for long term discussion or larger issues.
```
Comment (rsyring) [inline CONTRIBUTING.rst]: Do we want FSA mail to go to flask@python.org?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+* Ask on `Stack Overflow`_. Search with Google first using:
+  ``site:stackoverflow.com flask-sqlalchemy {search term, exception message, etc.}``
+
+.. _Stack Overflow: https://stackoverflow.com/questions/tagged/flask-sqlalchemy?sort=linked
+
+Reporting issues
```
Comment (rsyring) [inline CONTRIBUTING.rst]: See also #728.  Do we want to duplicate content or just reference the issue template from here?

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  us identify the issue. This also helps check that the issue is not with your
+  own code.
+- Describe what actually happened. Include the full traceback if there was an
+  exception.
+- List your Python, Flask, and Werkzeug versions. If possible, check if this
+  issue is already fixed in the repository.
+
+.. _minimal, complete, and verifiable example: https://stackoverflow.com/help/mcve
+
+Submitting patches
```
Comment (rsyring) [inline CONTRIBUTING.rst]: Should we encourage people to open issues for discussion before jumping in with a PR?

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+------------------
+
+- Include tests if your patch is supposed to solve a bug, and explain
```
Comment (rsyring) [inline CONTRIBUTING.rst]: How about **Always include tests.**  :)

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+- Create a virtualenv::
+
+        python3 -m venv env
+        . env/bin/activate
+        # or "env\Scripts\activate" on Windows
+
+- Install Flask-SQLAlchemy in editable mode with development dependencies::
+
+        pip install -e ".[dev]"
```
Comment (rsyring) [inline CONTRIBUTING.rst]: We don't have dev dependencies specified yet, but I think it's a good idea to make that happen.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (ThiefMaster) [reply CONTRIBUTING.rst]: Do we want/need a mailing list at all? I just looked at the archive (I'm not even subscribed myself..) and the Flask list is incredibly low-volume.

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+Thank you for considering contributing to Flask-SQLAlchemy!
+
+Support questions
+-----------------
+
+Please, don't use the issue tracker for this. Use one of the following
+resources for questions about your own code:
+
+* The IRC channel ``#pocoo`` on FreeNode.
```
Comment (ThiefMaster) [inline CONTRIBUTING.rst]: s/FreeNode/Freenode

Comment (ThiefMaster) [reply CONTRIBUTING.rst]: I'd add the discord link as well.

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+.. _Fork: https://github.com/pallets/flask-sqlalchemy/fork
+.. _Clone: https://help.github.com/articles/fork-a-repo/#step-2-create-a-local-clone-of-your-fork
+
+Start coding
+~~~~~~~~~~~~
+
+- Create a branch to identify the issue you would like to work on (e.g.
+  ``2287-dry-test-suite``)
+- Using your favorite editor, make your changes, `committing as you go`_.
+- Try to follow `PEP8`_, but you may ignore the line length limit if following
```
Comment (rsyring) [inline CONTRIBUTING.rst]: Duplicate content from submitting a patch.  Can we make this document more DRY?

File: CONTRIBUTING.rst [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+.. _create a pull request: https://help.github.com/articles/creating-a-pull-request/
+
+.. _contributing-testsuite:
+
+Running the tests
+~~~~~~~~~~~~~~~~~
+
+Install the testing requirements::
+
+    pip install pytest mock tox coverage
```
Comment (rsyring) [inline CONTRIBUTING.rst]: `pip install -e ".[tests]"`?

**Comment by rsyring** [comment]
I added some comments to this PR, but @davidism has been particular about how the docs & readme look, so I'll leave it up to him to approve or not.

@lbeaufort thanks for your contributions!

**Comment by rsyring** [comment]
We would probably want this PR re-targeted to the `2.x` branch.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (lbeaufort) [reply CONTRIBUTING.rst]: I confirmed this with @davidism but I'm happy to change it if you all decide to go in a different direction.

File: CONTRIBUTING.rst (hunk 2 above)
Comment (lbeaufort) [reply CONTRIBUTING.rst]: This mirrors the language in [Flask's contributing guidelines](https://github.com/pallets/flask/blob/master/CONTRIBUTING.rst#reporting-issues). Happy to make any changes here - if you all decide to remove it, should I put in a PR to Flask to remove it there too?

File: CONTRIBUTING.rst (hunk 5 above)
Comment (lbeaufort) [reply CONTRIBUTING.rst]: I put in and then closed this issue https://github.com/pallets/flask-sqlalchemy/issues/725 to update `setup.py` after speaking with @davidism. Should this instead say ` pip install -e .`? I

Comment (lbeaufort) [reply CONTRIBUTING.rst]: I went ahead and updated it to `pip install -e .`

File: CONTRIBUTING.rst (hunk 8 above)
Comment (lbeaufort) [reply CONTRIBUTING.rst]: See above - I don't think there's a `tests` extra in `setup.py` for this project.

**Comment by lbeaufort** [comment]
> We would probably want this PR re-targeted to the `2.x` branch.

@rsyring because this isn't a code change, it might make sense to target `master` otherwise no one will be able to see it until `2.x` gets merged into master. Does that happen often?

**Comment by rsyring** [comment]
> @rsyring because this isn't a code change, it might make sense to target master otherwise no one will be able to see it until 2.x gets merged into master. Does that happen often?

Good point.  Although, I wonder if contributing info should also be in the docs.  Does Flask or Werkzeug (which @davidism has often pointed me to for examples) ever include contributing content in their docs?

I'm sorry I can't be more definitive on suggestions for the docs or replies to your other comments in the PR.  I tried to make changes to docs/readme previously and @davidism really seemed to want to keep things consistent with other Pallets projects, but I'm not familiar enough with those projects to grok exactly what is and is not ok.  And the default way that I would do things seems to have been different enough to get changed, so I'm kind of waiting for someone  else to come along and give you direction.  I know that can be a frustrating place to be in, sorry for that.

**Comment by davidism** [comment]
Oops, finally got around to this, sorry for letting it sit for so long and thanks for working on it.

I rewrote the contributing doc based on the latest instructions, and also took the opportunity to do a general review on the document and tweak some sections. The issue template was massive, so I moved most of that to the contributing doc.

**Pull request merged**